client/viewer/js/main.js

```diff
@@ -1,75 +1,95 @@
 /* eslint-env browser */
-/* global console, io*/
-const viewer = require('./viewer.js');
-const port = location.port ? ':'+location.port : '';
-const socket = io('//'+location.hostname+port+'/screens');
-
-console.log('Initialising socket.io...');
+'use strict';
 
-socket.on('connect', function() {
-	viewer.setConnectionState(true);
-});
-
-socket.on('disconnect', function() {
-	socket.io.reconnect();
-	viewer.setConnectionState(false);
-});
+const Viewer = require('ftlabs-screens-viewer');
+const Carousel = require('ftlabs-screens-carousel');
+const port = location.port ? ':'+location.port : '';
+const host = '//'+location.hostname+port;
 
-socket.on('heartbeat',function() {
-	setTimeout(function() {
-		socket.emit('heartbeat');
-	}, 3000);
-});
+function viewerIsRunningInElectron() {
+	return (navigator.userAgent.indexOf('Electron') > 0 && navigator.userAgent.indexOf('FTLabs-Screens') > 0);
+}
 
-socket.on('reload', function(){
-	window.location.reload();
-});
+// Called by the script loader once the page has loaded
+window.screensInit = function() {
```

Comment:
We'd need to do it anyway as it is part of the original script loading code.

Also the viewer does not handle any DOM it is a general interface for the websockets api.